public/js/app.js

```diff
@@ -0,0 +1,210 @@
+'use strict';
+
+let navigationButtons = document.getElementsByClassName('navButtons');
+for (let i = 0; i < navigationButtons.length; i++) {
+  navigationButtons[i].addEventListener('click', swapSelected);
+}
+
+function swapSelected() {
+  let selected = this.innerHTML;
```

Comment:
Instead of relying on the content of the spans, try and use an attribute of the html element to hold which subreddit should be loaded.